pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -52,8 +52,16 @@ func (r *RedisOptionsProvider) GetRedisOptions() (*component.RedisOptions, error
 	r.options.BackendCommonLabels = r.backendCommonLabels()
 	r.options.BackendRedisLabels = r.backendRedisLabels()
 	r.options.BackendRedisPodTemplateLabels = r.backendRedisPodTemplateLabels()
-	r.options.SystemRedisPodTemplateAnnotations = r.apimanager.Spec.System.RedisAnnotations
-	r.options.BackendRedisPodTemplateAnnotations = r.apimanager.Spec.Backend.RedisAnnotations
+
+	var err error
+	r.options.SystemRedisPodTemplateAnnotations, err = r.systemRedisPodTemplateAnnotations()
+	if err != nil && !k8serr.IsNotFound(err) {
```

Comment:
what does this mean for a fresh install?

I mean, when redis has not been deployed yet, this code will not found the configmap. So, the podtemplate will not have the annotation. Then, in a subsequent reconciliation loop, pod template will be added, trigerring a new redeployment of the redis pod. Am I correct? In that case, not a big issue, but can we implement this better? The logic should be, the the operator creates the config map and waits until it finds it to created to create the deployment.